src/main/scala/is/hail/sparkextras/OrderedRDD.scala

```diff
@@ -194,13 +194,7 @@ object OrderedRDD {
     import kOk._
 
     import Ordering.Implicits._
-
-    rdd.partitioner match {
```

Comment:
violating this assertion means an invalid RDD was generated at some point, and we want to catch that code.